modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -2235,10 +2263,25 @@ private CompletableFuture<Void> handleChangePendingAssignmentEvent(
                         return;
                     }
 
+                    // A node might exist in stable yet we don't want to start the replica
+                    // The case is: nodes A, B and C hold a replication group,
+                    // nodes A and B die.
+                    // Reset adds C into force pending and user writes data onto C.
+                    // Then A and B go back online. In this case
+                    // stable = [A, B, C], pending = [C, force] and only C should be started.
+                    if (isRecovery && !replicaMgr.isReplicaStarted(replicaGrpId)) {
```

Comment:
Not sure that I understand:
- How we guarantee that this check will not break the non-HA case?
- This piece of code is not start any nodes, but just update the configuration of raft client, shouldn't we rewrite comment a little bit?